recipes/gmp/all/conanfile.py

```diff
@@ -100,37 +96,37 @@ def generate(self):
         tc = AutotoolsToolchain(self)
         yes_no = lambda v: "yes" if v else "no"
         tc.configure_args.extend([
-            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
-            "--enable-assembly={}".format(yes_no(not self.options.get_safe("disable_assembly", False))),
-            "--enable-fat={}".format(yes_no(self.options.get_safe("enable_fat", False))),
-            "--enable-cxx={}".format(yes_no(self.options.enable_cxx)),
-            "--srcdir={}".format(self.source_folder.replace("\\", "/")),
+            f'--with-pic={yes_no(self.options.get_safe("fPIC", True))}',
+            f'--enable-assembly={yes_no(not self.options.get_safe("disable_assembly", False))}',
+            f'--enable-fat={yes_no(self.options.get_safe("enable_fat", False))}',
+            f'--enable-cxx={yes_no(self.options.enable_cxx)}',
+            f'--srcdir={"../src"}', # Use relative path to avoid issues with #include "$srcdir/gmp-h.in" on Windows
         ])
+        env = tc.environment()
```

Comment:
please move back this line where it was before, it's very important because it captures the all environment induced by manipulations of toolchain flags, and you don't want to capture this state before manipulations done in msvc branch like extra_cflags etc.

Reply:
> please move back this line where it was before, it's very important because it captures the all environment induced by manipulations of toolchain flags, and you don't want to capture this state before manipulations done in msvc branch like extra_cflags etc.

Hi @SpaceIm,
Can you further clarify your concern here? In looking at the code for AutotoolsToolchain::environment(), I don't see it doing anything with extra_cflags, and even if it did, I don't understand why the various configuration options for msvc were being (what appears to be me to be arbitrarily) split. Thanks